ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/NoUnusedImportsRuleTest.kt

```diff
@@ -316,4 +441,22 @@ class NoUnusedImportsRuleTest {
             )
         )
     }
+
+    @Test
+    fun shouldNotReportUnusedWhenStaticImportIsFromAnotherPackage() {
+        assertThat(
+            NoUnusedImportsRule().lint(
+                """
+                package com.foo
+
+                import android.text.Spannable
+                import androidx.core.text.toSpannable
+
+                fun foo(text: String): Spannable {
+                    return text.toSpannable()
```

Comment:
This is kind of a funny case because I realized that when I renamed the `text` parameter of `foo` to anything else, lint no longer fired. In this case it's not even a static import, it's just a function call with a variable that happens to match the name of an extension functino.

Reply:
Yes,  since there was import ending with `text.toSpannable`, the code asssumed it to be a static import. 😞 
In this PR, in addition to the prev fixes, just made sure parent import (with full package name check) exists.